### ISSUE TITLE
[adapter-fetch, cypress] error on using fetch on app window

### ISSUE THREAD

**Issue by Cherviakov** [Netflix/pollyjs#354]
## Prerequisites
node: 14.4.0
os: 4.15.0-107-generic #108~16.04.1-Ubuntu
pollyjs: 5.0.0
cypress: 4.11.0

## Description
Trying to use pollyjs with cypress for e2e tests, using xhr and fetch adapters,
when trying to make call from cypress window fetch it works but, obviously,
not get recorded, when call fetch from application window get error from 
pollyjs fetch adapter, seems from call to serialize response headers:

First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object

Server logs shows that request made successfully and response sent.

### Shareable Source

```js
 import { Polly } from '@pollyjs/core';
 import XHRAdapter from '@pollyjs/adapter-xhr';
 import FetchAdapter from '@pollyjs/adapter-fetch';
 import RESTPersister from '@pollyjs/persister-rest';
 
 Polly.register(XHRAdapter);
 Polly.register(FetchAdapter);
 Polly.register(RESTPersister);
 
 describe('ui test', () => {
   it('title present', () => {
     cy.visit('/', {
       onBeforeLoad: (win) => {
         win.polly = new Polly('Cypress-Tooth', {
           adapters: ['xhr', fetch'],
           adapterOptions: {
             xhr: {
               context: win,
             },
             fetch: {
               context: win,
             }
           },
           persister: ['rest'],
           persisterOptions: {
             rest: {
               host: 'http://localhost:3002'
             }
           },
         });
       }
     });
     cy.wait(2000);
     cy.get('h1').should('be.visible');
     cy.url();
     cy.window().invoke('fetch', 'http://localhost:3000').then((response) => {
       expect(response.status).to.be.equal(200);
     })
  });
  afterEach(() => {
     cy.url();
     cy.window().its('polly').invoke('stop');
   });
 });


```

### Error Message & Stack Trace

```
First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object
```

### Config

Copy the config used to setup the Polly instance:

```js
new Polly('Recording Name', {
      adapters: ['xhr', fetch'],
           adapterOptions: {
             xhr: {
               context: win,
             },
             fetch: {
               context: win,
             }
           },
           persister: ['rest'],
           persisterOptions: {
             rest: {
               host: 'http://localhost:3002'
             }
           },
});
```

### Dependencies

Copy the @pollyjs dependencies from `package.json`:

```json
{
  "@pollyjs/core": "5.0.0",
  "@pollyjs/adapter-x": "5.0.0",
  "@pollyjs/persister-x": "5.0.0"
}
```

## Relevant Links

- If your project is public, link to the repo so we can investigate directly.
- **BONUS POINTS:** Create a [minimal reproduction](http://stackoverflow.com/help/mcve) and upload it to GitHub. This will get you the fastest support.

## Environment

Tell us which operating system you are using, as well as which versions of Node.js and npm/yarn. If applicable, include the browser and the corresponding version.

Run the following to get it quickly:

```
linux 4.15.0-107-generic
6.4.15
```

**Comment by offirgolan** [comment]
Apologies for the delayed response. Is there anyway you create a reproduction for me to tinker with? Is this happening just on the `fetch` or `xhr` adapter or both?

**Comment by Cherviakov** [comment]
It happened for fetch adapter only. On simple example code I have though that bug not reproduced, so I guess have to close this one.

**Comment by nathanqueija** [comment]
I'm also getting the same error with the exact same configuration you have @Cherviakov . DId you manage to solve the problem with the fetch adapter?

**Comment by ulrik59** [comment]
I have exactly the same problem with `@pollyjs/adapter-fetch >= 4.3.0`. @Cherviakov did you succeed to solve this problem? If you did, how?

**Comment by ulrik59** [comment]
@Cherviakov @offirgolan I may have found the reason of the problem. The bug was introduce in `@pollyjs/adapter-fetch@4.3.0`. When setting `window` as `context` for fetch adapter, the `nativeFetch` method is the one from the `window` object and calling `.arrayBuffer()` on the response of the fetch is resulting in a `ArrayBuffer` object but from the `window` context not the node context.
So in the `passthroughRequest` method (see https://github.com/Netflix/pollyjs/blob/master/packages/%40pollyjs/adapter-fetch/src/index.js#L156), this will give the following result:
```js
console.log((await response.arrayBuffer()) instanceof context.ArrayBuffer) // Will return true . "context" here in the window object
console.log((await response.arrayBuffer()) instanceof ArrayBuffer) // Will return false
```
This is why doing:
```js
const buffer = Buffer.from(await response.arrayBuffer());
```
is resulting in this error:
```
First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object
```
Because the NodeJS method `Buffer.from` can't identify the result of `response.arrayBuffer()` as a valid NodeJS `ArrayBuffer`.
Unfortunately I have no idea how to fix that. Locking `@pollyjs/adapter-fetch` to `4.2.1` even if all others `@pollyjs/*` deps are in 5+ is working for me.

**Comment by Cherviakov** [comment]
Thanks for investigating this one, I found it accidentally while struggling with making polly work with cypress, just tried fetch as though it may give better results and reported this problem occured to me. As it not reproduced for me later and my problem was not exactly about using fetch I just left it as it is.

**Comment by offirgolan** [comment]
Fixed in #393.
